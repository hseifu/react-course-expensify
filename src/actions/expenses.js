import uuid from 'uuid';

//ADD EXPENSE
export const addExpense = (
    { 
        description = '',
        note = '', 
        amount = 0, 
        createdAt = 0
    } = {}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

//Remove Expense
export const removeExpense = (removeId) => ({
    type: 'REMOVE_EXPENSE',
    removeId
})

//Edit Expense
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})