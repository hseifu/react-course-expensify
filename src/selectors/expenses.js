import moment from 'moment';

//get visible data
export default (expenses, {text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment,'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment,'day') : true;
        const reg = new RegExp('^'+text.toLowerCase()+'|\\s'+text.toLowerCase(),'g');
        const textMatch = expense.description.toLowerCase().match(reg);
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else {
            return a.amount < b.amount ? 1 : -1
        }
    })
}