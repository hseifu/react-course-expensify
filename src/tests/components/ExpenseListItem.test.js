import React from 'react';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test("should render expense item passed in ", () => {
    const wrapper = shallow(<ExpenseListItem expense={expenses[1]}/>)
    expect(wrapper).toMatchSnapshot();
})