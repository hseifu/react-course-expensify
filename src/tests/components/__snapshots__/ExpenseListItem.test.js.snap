// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item passed in  1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  >
     
    <h3 />
     
  </Link>
  <p>
    undefined undefined
  </p>
</div>
`;
